independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProcessor.java

```diff
@@ -0,0 +1,159 @@
+package io.quarkus.devtools.codestarts;
+
+import static io.quarkus.devtools.codestarts.Codestart.BASE_LANGUAGE;
+
+import io.quarkus.devtools.codestarts.reader.CodestartFile;
+import io.quarkus.devtools.codestarts.reader.CodestartFileReader;
+import io.quarkus.devtools.codestarts.strategy.CodestartFileStrategy;
+import io.quarkus.devtools.codestarts.strategy.CodestartFileStrategyHandler;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class CodestartProcessor {
+
+    private final CodestartResourceLoader resourceLoader;
+    private final String languageName;
+    private final Path targetDirectory;
+    private final List<CodestartFileStrategy> strategies;
+    private final Map<String, Object> data;
+    private final Map<String, List<CodestartFile>> files = new LinkedHashMap<>();
+
+    CodestartProcessor(final CodestartResourceLoader resourceLoader,
+            final String languageName,
+            final Path targetDirectory,
+            List<CodestartFileStrategy> strategies,
+            final Map<String, Object> data) {
+        this.resourceLoader = resourceLoader;
+        this.languageName = languageName;
+        this.targetDirectory = targetDirectory;
+        this.strategies = strategies;
+        this.data = data;
+    }
+
+    void process(final Codestart codestart) throws IOException {
+        resourceLoader.loadResourceAsPath(codestart.getResourceDir(), p -> {
+            final Path baseDir = p.resolve(BASE_LANGUAGE);
+            final Path languageDir = p.resolve(languageName);
+            if (!codestart.getSpec().getMissingLanguages().isEmpty()
+                    && codestart.getSpec().getMissingLanguages().contains(languageName)) {
+                // TODO print a warning to inform that this codestart miss this language
```

Comment:
that todo seems important to fix?

Reply:
yes, I am not sure yet how I will deal with those. Recently we found out that tooling could prefer a `Result` more than some "random" printed messages.

Reply:
not sure I follow what you mean it prefer a Result ?

Reply:
Well, when implementing the raw CLI, we discussed with @stalep that it was hard to produce an interactive experience because the common tooling was producing logs, but the consumer, didn't know about it beside the fact that it failed.

Reply:
I'll do that in another PR

Reply:
https://github.com/quarkusio/quarkus/issues/11084